dbschema/user.esdl

```diff
@@ -33,5 +33,26 @@ module default {
 }
  
 module User {
+  type Education extending default::Resource {
+    required degree: Degree;
+    required major: str;
+    required institution: str;
+  }
+
+  type Unavailability extending default::Resource {
+    required description: str;
+    required start: cal::local_date;
+    required end: cal::local_date;
```

Comment:
Looks like start/end are reserved words. So...
```suggestion
    required `start`: cal::local_date;
    required `end`: cal::local_date;
```
These backticks will be required for use in EdgeQL as well. I don't think this is that big of a deal. Data Explorer and Query Builder handle these normally. EdgeQL will probably use `{*}` for select and only need to specify on insert query.

I don't think it's worth changing the property names. I'd stick with these and use backticks 👍 

---

Another thought here is to use a range instead.
```
range<cal::local_date>
```
This would verify we have a valid range automatically, otherwise we'd need to [verify this constraint ourselves](https://github.com/SeedCompany/cord-api-v3/blob/cf9b844877dfed414124754ee369f06242af5180/dbschema/project.esdl#L26-L26).

Not sure about which way would be more idiomatic.

Reply:
Honestly, I really like the idea of `range<cal::local_date>`.  It's really exactly what we need here, built-in constraint and all.  Good idea.